File: since2020/CodeForces/1312C.py
'''
	@judge CodeForces
	@id 1312C
	@name Adding Powers

	@tag Greedy, Number Base Conversion
'''
from math import ceil, log
from sys import stdin

def toKBase(n, k, d):
	for _ in range(d + 1):
		yield n % k
		n //= k

def solve(arr, k):
	if max(arr) == 0:
		return True
	digit = ceil(log(max(arr), k))
	return all(map(lambda ls: sum(ls) <= 1, zip(*map(lambda n: tuple(toKBase(n, k, digit)), arr))))

input()
for fst, nxt in zip(*[ iter(stdin) ] * 2):
	_, k = map(int, fst.split())
	arr  = list(map(int, nxt.split()))
	print('YES' if solve(arr, k) else 'NO')

File: since2020/CodeForces/1312C.cpp
/**
 *  @judge CodeForces
 *  @id 1312C
 *  @name Adding Powers
 *  @contest Educational Codeforces Round 83
 * 
 *  @tag Greedy, Number Base Conversion
 */
#include <bits/stdc++.h>
using namespace std;

const int MAX = 110;

long long n, k, kpower[MAX];
vector<long long> a;

void intoKBase(long long x) {
	for (auto &&kp : kpower) {
		if (x <= 0)
			return ;
		kp += x % k;
		x /= k;
	}
	return ;
}

bool solve() {
	for (auto && kp : kpower)
		kp = 0;
	for (const auto &x : a)
		intoKBase(x);
	for (const auto &kp : kpower)
		if (kp > 1)
			return false;
	return true;
}

int main() {
	int t;
	for (cin >> t; t; t--) {
		cin >> n >> k;
		a = vector<long long>(n, 0);
		for (auto && x : a)
			cin >> x;
		cout << (solve() ? "YES" : "NO") << endl;
	}
}